phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexScrutinyToolPerformanceIT.java

```diff
@@ -0,0 +1,219 @@
+        }
+    }
+
+    @Test
+    public void testSkipScan() throws Exception {
+        String baseTableName = dataTableName;
+        PreparedStatement upsertStmt = dataTableUpsertStmt;
+        if (!Strings.isNullOrEmpty(viewDdl)) {
+            baseTableName = viewName;
+            upsertStmt = viewUpsertStmt;
+        }
+
+        for (int i=0; i < 2; i++) {
+            if (dataTableDdl.contains("MULTI_TENANT")) {
+                upsertRowMultiTenant(upsertStmt, "val"+i, i,"name-" + i, 19999);
+            } else {
+                upsertRow(upsertStmt, i, "name-" + i, 19999);
+            }
+            conn.commit();
+        }
+
+        IndexScrutinyToolIT.runScrutiny(schemaName, baseTableName,
+                indexTableName, null, SourceTable.INDEX_TABLE_SOURCE);
+        PreparedStatement targetStmt = IndexScrutinyTool.getTargetTableQueryForScrutinyTool();
+        assertSkipScanFilter(targetStmt);
+
+        IndexScrutinyToolIT.runScrutiny(schemaName, baseTableName,
+                indexTableName, null, SourceTable.DATA_TABLE_SOURCE);
+        targetStmt = IndexScrutinyTool.getTargetTableQueryForScrutinyTool();
+        assertSkipScanFilter(targetStmt);
+    }
+
+    private void assertSkipScanFilter(PreparedStatement targetStmt) throws SQLException {
+        QueryPlan plan = targetStmt.unwrap(PhoenixStatement.class).getQueryPlan();
+        Scan scan = plan.getScans().get(0).get(0);
+        Filter filter = scan.getFilter();
+        if (filter instanceof FilterList) {
+            FilterList filterList = (FilterList) scan.getFilter();
+
+            boolean isSkipScanUsed = false;
+            for (Filter f : filterList.getFilters()) {
+                if (f instanceof SkipScanFilter) {
+                    isSkipScanUsed = true;
+                    break;
+                }
+            }
+            assertEquals(true, isSkipScanUsed);
+        } else {
+            assertEquals(SkipScanFilter.class, filter.getClass());
+        }
+    }
+    
+    private void generateUniqueTableNames() {
+        schemaName = "S_" + generateUniqueName();
+        dataTableName = "D_" + generateUniqueName();
+        dataTableFullName = SchemaUtil.getTableName(schemaName, dataTableName);
+        viewName = "V_" + generateUniqueName();
+        viewFullName = SchemaUtil.getTableName(schemaName, viewName);
+        indexTableName = "I_" + generateUniqueName();
+        indexTableFullName = SchemaUtil.getTableName(schemaName, indexTableName);
+    }
+
+    private void upsertRow(PreparedStatement stmt, int id, String name, int zip) throws SQLException {
+        int index = 1;
+        // insert row
+        stmt.setInt(index++, id);
+        stmt.setString(index++, name);
+        stmt.setInt(index++, zip);
+        stmt.setTimestamp(index++, new Timestamp(testTime));
+        stmt.executeUpdate();
+    }
+
+    private void upsertRowMultiTenant(PreparedStatement stmt, String col, int id, String name, int zip) throws SQLException {
```

Comment:
Can this be moved to IndexScrutinyToolBaseIT and reused in both ITs, this and IndexScrutinyToolIT?

Reply:
IndexScrutinyToolIT doesn't have multi tenant test. There is a separate test file for multi tenant but it is a bigger change. It doesn't use prepared statement and uses tenant connections and global connections.